src/ImageProcessor.Web.Plugins.PostProcessor/PostProcessor.cs

```diff
@@ -53,13 +53,11 @@ public static async Task<MemoryStream> PostProcessImageAsync(HttpContext context
                 // Save source file length
                 var length = stream.Length;
 
-                // Create a temporary source file with the correct extension
-                var tempSourceFile = Path.GetTempFileName();
-                sourceFile = Path.ChangeExtension(tempSourceFile, extension);
-                File.Move(tempSourceFile, sourceFile);
-
-                // Give our destination file a unique name
-                destinationFile = sourceFile.Replace(extension, "-out" + extension);
+                // Get temporary file names
+                var tempPath = Path.GetTempPath();
+                var tempFile = Path.GetFileNameWithoutExtension(Path.GetRandomFileName());
```

Comment:
Instead of using [`Path.GetTempFileName()`](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename) that creates predictable file names (and has a 65k file limit), this is using [`Path.GetRandomFileName()`](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getrandomfilename) to generate a cryptographically strong, random file name.